network/src/peer_manager/mod.rs

```diff
@@ -816,24 +829,44 @@ where
     ) {
         let mut upstream_handlers = self.upstream_handlers.clone();
         let network_context = self.network_context.clone();
+        let inbound_rate_limiter = self.inbound_rate_limiter.clone();
         self.executor.spawn(network_events.for_each_concurrent(
             self.max_concurrent_network_reqs,
             move |inbound_event| {
                 Self::handle_inbound_event(
                     network_context.clone(),
                     inbound_event,
                     peer_id,
+                    inbound_rate_limiter.clone(),
                     &mut upstream_handlers,
                 );
                 futures::future::ready(())
             },
         ));
     }
 
+    fn rate_limit<'a>(
+        inbound_rate_limiter: &'a RateLimiter<PeerId, DefaultKeyedStateStore<PeerId>, DefaultClock>,
+        peer_id: &PeerId,
+        msg_length: usize,
+    ) -> Result<(), NegativeMultiDecision<NotUntil<'a, QuantaInstant>>> {
+        // We can't rate limit something of 0 size
+        if msg_length < 1 {
+            return Ok(());
+        }
+        let length = NonZeroU32::new(msg_length as u32).unwrap();
```

Comment:
why unwrap in something that returns a result?

Reply:
This is required in the next line to not be a result?